TF-IDF API

DT9 opened the pull request:
Hey Gary, can you look over our code? 
This is the tentative API for TFIDF where it queries from publications and outputs the scores.

GarryBullock commented on web/bfex/components/keyword/keygen.py:
```diff
@@ -0,0 +1,45 @@
+        }
+
+        publication = es.termvectors(
+            index="Publication", doc_type="doc", body=body)
```
Take a look at the Publication model. This isn't the correct index.

GarryBullock commented:
So I think overall this is a good start. But this should not be an api. It should be a keyword generating approach, as defined in the high level API. Lets move this functionality under web/bfex/components/key_generation/...

You'll need to implement the KeyGenerator and KeyGenerationApproach classes, then TFIDF should exist as an implementation of KeyGenerationApproach.

Use AbstractBaseClass (import ABC) in KeyGenerationApproach to get something similar to an interface in python.

DT9 commented:
Implemented KeyGenerator and KeyGenerationApproach from the class diagrams.
Removed TFIDF & elastic related code until I get some clarity on information retrieval concepts tomorrow.
Instead created a generic word frequency analysis without stopwords.
